torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -582,13 +588,19 @@ def batched_powerSGD_hook(
     # Apply PowerSGD after `start_powerSGD_iter` iterations.
     device = input_tensor.device
     total_length = input_tensor.shape[0]
+    state.total_numel_before_compression += total_length
 
     # View the input tensor as a 2D square-shape tensor, and pad 0s if necessary.
     square_side_length = math.ceil(math.sqrt(total_length))
+    state.total_numel_after_compression += (
```

Comment:
Even though this is the exact shape for P and Q respectively, I wonder if it'll be more readable if we set `total_numel_after_compression` after we've explicitly computed P and Q and using its shape. This seems similar to what you do in `powersgd_hook`.